BurnOutSharp/ProtectionType/Tages.cs

```diff
@@ -56,74 +55,132 @@ public string CheckContents(string file, byte[] fileContent, bool includeDebug,
-            {
-                return "TAGES";
-            }
-            else if (Path.GetFileName(path).Equals("tagesclient.exe", StringComparison.OrdinalIgnoreCase))
+            var matchers = new List<PathMatchSet>
             {
-                return "TAGES Activation Client " + Utilities.GetFileVersion(path);
-            }
-            else if (Path.GetFileName(path).Equals("TagesSetup.exe", StringComparison.OrdinalIgnoreCase))
-            {
-                return "TAGES Setup " + Utilities.GetFileVersion(path);
-            }
-            else if (Path.GetFileName(path).Equals("TagesSetup_x64.exe", StringComparison.OrdinalIgnoreCase))
+                new PathMatchSet(new PathMatch("Devx.sys", useEndsWith: true), "TAGES Driver"),
+                new PathMatchSet(new PathMatch("VtPr.sys", useEndsWith: true), "TAGES Driver"),
+                new PathMatchSet(new PathMatch("ENODPL.VXD", useEndsWith: true), "TAGES 9x Driver"),
+                new PathMatchSet(new PathMatch("TANDPL.VXD", useEndsWith: true), "TAGES 9x Driver"),
+                new PathMatchSet(new PathMatch("enodpl.sys", useEndsWith: true), "TAGES NT Driver"),
+                new PathMatchSet(new PathMatch("tandpl.sys", useEndsWith: true), "TAGES NT Driver"),
+                new PathMatchSet(new PathMatch("hwpsgt.vxd", useEndsWith: true), "TAGES 9x Driver"),
+                new PathMatchSet(new PathMatch("lemsgt.vxd", useEndsWith: true), "TAGES 9x Driver"),
+                new PathMatchSet(new PathMatch("hwpsgt.sys", useEndsWith: true), "TAGES NT Driver"),
+                new PathMatchSet(new PathMatch("lemsgt.sys", useEndsWith: true), "TAGES NT Driver"),
+
+                // The following files are supposed to only be found inside the driver setup executables.
+                new PathMatchSet(new PathMatch("ithsgt.sys", useEndsWith: true), "TAGES Driver"),
+                new PathMatchSet(new PathMatch("lilsgt.sys", useEndsWith: true), "TAGES Driver"),
+                new PathMatchSet(new PathMatch("atksgt.sys", useEndsWith: true), "TAGES Driver"),
+                new PathMatchSet(new PathMatch("lirsgt.sys", useEndsWith: true), "TAGES Driver"),
+
+                // The following files appear to be container formats for TAGES, but little is currently known about them.
+                new PathMatchSet(new PathMatch("GameModule.elb", useEndsWith: true), "TAGES/SolidShield Game Executable Container"),
+                new PathMatchSet(new PathMatch("InstallModule.elb", useEndsWith: true), "TAGES/SolidShield Installer Container"),
+
+                // Not much is known about this file, but it seems to be related to what PiD reports as "protection level: Tages BASIC". Seems to always be found with other KWN files.
+                new PathMatchSet(new PathMatch("InstallModule.elb", useEndsWith: true), "TAGES (BASIC?)"),
+            };
+
+
+            else if (Path.GetFileName(path).Equals("GAME.KWN", StringComparison.OrdinalIgnoreCase))
```

Comment:
This is broken because you didn't migrate this check. It should be failing with an issue that says there's no matching `if` or something similar.